pkg/operator/discover/discover.go

```diff
@@ -241,6 +239,13 @@ func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, disco
 
 }
 
+func getLabels() map[string]string {
+	labels := make(map[string]string)
+	k8sutil.AddRecommendedLabels(labels, "rook-discover", "rook-ceph-operator", "rook-discover", "rook-discover")
```

Comment:
This is probably the only place in operator code I'd expect to hard code the operator to be the parent.